### ISSUE TITLE
pgSQL & Redis external - Error executing 'postInstallation': Files to substitute resolved to an empty list

### ISSUE THREAD

**Issue by lejeczek** [bitnami/bitnami-docker-discourse#183]
**Description**
Hi.
I deploy with:

-> $ podman run -d --pod=some.tet-${HOSTNAME%%.*} --name some.tet-discourse --restart=always -e DISCOURSE_USERNAME=adm -e DISCOURSE_PASSWORD=some#PASS -e DISCOURSE_EMAIL=me@some.tet -e DISCOURSE_SITENAME=some.tet -e DISCOURSE_HOSTNAME=some.tet -e POSTGRESQL_ROOT_USER=discO -e POSTGRESQL_ROOT_PASSWORD=some#PASS -e POSTGRESQL_HOST=10.88.2.1 -e DISCOURSE_POSTGRESQL_USERNAME=discO -e DISCOURSE_POSTGRESQL_PASSWORD=some#PASS -e DISCOURSE_POSTGRESQL_NAME=discO -e REDIS_HOST=10.88.2.1 -e REDIS_PASSWORD=some#PASS --volume /srv/containers/FLATfiles/some.tet.discourse:/bitnami:z docker.io/bitnami/discourse

This is my first foray into Bitnami's thus I try the whole lot, not being sure 100% what is what.
Containers fails to boostrap, in logs I see:
...
discour INFO  Patching Discourse...
postgre INFO  Trying to connect to PostgreSQL server
postgre INFO  Found PostgreSQL server listening at 10.88.2.1:5432
postgre INFO  PostgreSQL server listening and working at 10.88.2.1:5432
discour INFO  Preparing Discourse environment...
discour INFO  Configuring the Discourse database files...
Error executing 'postInstallation': Files to substitute resolved to an empty list

At pgSQL logs end, during container boot/start:
...
2021-04-28 14:13:07.237 EDT [2825844] DETAIL:  Password does not match for user "discO". 
  Connection matched pg_hba.conf line 26: "host    discO  discO      10.88.2.0/24           md5"

Which is weird as I can login with those credentials, with that password, via 'psql' cmd.

One thing, which confuses me are "ROOTs" for pgSQL - I assumed that those of us who want to use remote/external pgSQL usually do not want to reveal, to give 'root' to the Discourse container, so I use db owner as root here (ps. those parts could be bit better documented by devel)
many thanks, L.

**Steps to reproduce the issue:**

1. [First Step]
2. [Second Step]
3. [and so on...]

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**
Redis 6.0.9
psql (13.2)

- Output of `docker version`:

```
(paste your output here)
```

- Output of `docker info`:

```
(paste your output here)
```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by lejeczek** [comment]
Could it be the '#' in the password which I did not escape?
It might be looking that way. Yet still, now without '#' and not complaints in pgSQL logs:
...
nami    INFO  Initializing postgresql-client
nami    INFO  postgresql-client successfully initialized
nami    INFO  Initializing discourse
discour INFO  Patching Discourse...
postgre INFO  Trying to connect to PostgreSQL server
postgre INFO  Found PostgreSQL server listening at 10.88.2.1:5432
postgre INFO  PostgreSQL server listening and working at 10.88.2.1:5432
discour INFO  Preparing Discourse environment...
discour INFO  Configuring the Discourse database files...
discour INFO  Configuring Redis...
discour INFO  Setting up SMTP in Discourse...
discour INFO  Configuring Discourse database...
discour INFO  Enabling PostgreSQL plugins for Discourse...
discour INFO  Migrating database...
Error executing 'postInstallation': Failed to report error: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? 2 ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? subscribe failed, reconnecting in 1 second. Call stack /opt/bitnami/discourse/vendor/bundle/ruby/2.6.0/gems/redis-4.2.5/lib/redis/client.rb:132:in `call'

**Comment by alvneiayu** [comment]
hi @lejeczek 

Yes, sometimes those kind of symbols generate problems with the passwords. Try to do a test using a password without any special symbols (just letters). I think that you are going to see the discourse running without any error.

Thanks and let me know if I can help you with something else.

Álvaro

**Comment by lejeczek** [comment]
@alvneiayu I do not quiet get your meaning. Did you read my own, second comment?

**Comment by alvneiayu** [comment]
hi @lejeczek 

Sorry, my fault. I didn't notice you last comment. I am trying to reproduce your problem. I will come back soon with some information.

Thanks

Álvaro

**Comment by alvneiayu** [comment]
hi @lejeczek 

I created a docker-compose file trying to reproduce your problem. I am not using an external postgresql (I am creating a new one) but it is working. Maybe this file can help you a little bit to face your problem.

[docker-compose.zip](https://github.com/bitnami/bitnami-docker-discourse/files/6414947/docker-compose.zip)

Let me know if I can help you with something else.

Álvaro

**Comment by lejeczek** [comment]
Hi @alvneiayu 

I do not know but I'll go ahead and assume that you are trying to help but I have to ask - how good help is that if any, when the problem when described specifically you do not address but instead you do something else.

That compose file is of no use to me nor anybody who might come across the problem I'm describing.
When I put it all into one pod then all works and the problem does not occur, but again - if I have to stress it - the issue is when one wants to use "external" Redis & pgSQL (or maybe just pgSQL).
For a developer - if you are one - the only work worth investing time should be to **reproduce** the problem exactly, which is not what you do, I think - and to reproduce this one should be simple: suffice to use pgSQL on the host itself, not in a container, which is what I'm doing, simple.

Only bit here in my setup which is different with regards to pgSQL, is - my pgSQL is ver 13.x whereas Bitnami's uses 11.x. and I'll try that version as soon as I can.

regards, L.

**Comment by alvneiayu** [comment]
hi @lejeczek 

I reproduced your problem:
```
Error executing 'postInstallation': Failed to report error: ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? 2 ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct? subscribe failed, reconnecting in 1 second. Call stack /opt/bitnami/discourse/vendor/bundle/ruby/2.6.0/gems/redis-4.2.5/lib/redis/client.rb:132:in `call'
```
Could you verify in your Redis configuration file that the element "requirepass" is set up in your redis configuration file (not commented)?

Looking in Google I found other users having the same problem with Redis:
https://github.com/chatwoot/chatwoot/issues/1618
https://help.nextcloud.com/t/solved-redisexception-err-auth-password-level-3-nextcloud-log-error/88175

Regards
Álvaro

**Comment by lejeczek** [comment]
Hi @alvneiayu 

Sure those errors line from log snippet I pasted does not help - since it follows pgSQL info - but I should have looked more closely at that log.
Well investigated you. I can add that it actually most likely has nothing to do with "requirepass" but rather with the fact that "ifaces" of the host did not "play" well enough with Redis on the host - I have wireguard devices tangled in the middle and for some reason Redis did not, while pgSQL did, pick up and start to listen on those wireguard devices.
To sum it up - almost certainly that 'Error executing 'postInstallation' was caused by Redis _not_ serving on that given IP (my case, but I'd imagine with external to the container servers it might be as well firewall, etc)
Deployment works with "requirepass"

many thanks, L.
